pkg/helpers/helpers.go

```diff
@@ -505,3 +509,18 @@ func GetComponentNamespace() (string, error) {
 	}
 	return string(nsBytes), nil
 }
+
+func GetNodeSelector(cluster *clusterv1.ManagedCluster) (map[string]string, error) {
+	// This part is to support running pods related local-cluster on specified nodes,like infra nodes.
+	nodeSelector := map[string]string{}
+	if cluster.GetName() == "local-cluster" {
+		annotations := cluster.GetAnnotations()
+		if nodeSelectorString, ok := annotations[AnnotationNodeSelector]; ok {
```

Comment:
I think a better way is always short circuit.

``` go
if cluster.GetName() != "local-cluster" {
  return nodeSelector
}

annotations := cluster.GetAnnotations()
if annotations == nil {
   return nodeSelector
}

nodeSelectorString, ok := annotations[AnnotationNodeSelector]
if !ok {
  return nodeSelector
}
```

Reply:
done